### PR TITLE
Support scale controller for StatefulSet

### PR DESCRIPTION
**Pull request by whichxjy**
This PR adds support for the StatefulSet scale controller. It's a part of #1653.

### CODE REVIEW

**Comment by wangyikewxgm** [comment]
Good first work. statefulset rollout controller will contained by next pr?

**Comment by whichxjy** [comment]
> Good first work. statefulset rollout controller will contained by next pr?

Yes, sure.

File: pkg/controller/common/rollout/rollout_plan_controller.go [hunk 1]
```diff
@@ -357,6 +359,15 @@ func (r *Controller) GetWorkloadController() (workloads.WorkloadController, erro
...
 		}
+
+		// check if the target workload is StatefulSet
+		if r.targetWorkload.GetKind() == reflect.TypeOf(apps.StatefulSet{}).Name() {
+			if r.sourceWorkload == nil {
+				return workloads.NewStatefulSetScaleController(r.client, r.recorder, r.parentController,
+					r.rolloutSpec, r.rolloutStatus, target), nil
+			}
+		}
 	}
```
Comment (wangyikewxgm) [inline pkg/controller/common/rollout/rollout_plan_controller.go:370]: Please delete. When next pr (`statefulset rollout controller` )ready we can add this back.

File: pkg/controller/common/rollout/workloads/statefulset_controller.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	if controller := metav1.GetControllerOf(statefulSet); controller != nil &&
+		controller.Kind == v1beta1.AppRolloutKind && controller.APIVersion == v1beta1.SchemeGroupVersion.String() {
+		// it's already there
+		return true, nil
+	}
+
+	statefulSetPatch := client.MergeFrom(statefulSet.DeepCopyObject())
+
+	// add the parent controller to the owner of the StatefulSet
+	ref := metav1.NewControllerRef(c.parentController, v1beta1.AppRolloutKindVersionKind)
```
Comment (wangyikewxgm) [inline pkg/controller/common/rollout/workloads/statefulset_controller.go:51]: please replace `v1beta1.AppRolloutKindVersionKind` with `c.parentController.GetObjectKind().GroupVersionKind()` we will support more kind of object  as parent

Comment (whichxjy) [reply pkg/controller/common/rollout/workloads/statefulset_controller.go:51]: It seems like `c.parentController.GetObjectKind().GroupVersionKind()` would return an empty `GroupVersionKind` struct:

```
GroupVersionKind{
    Group:   "",
    Version: "",
    Kind:    "",
}
```

Is there another way to do that?

Comment (wangyikewxgm) [reply pkg/controller/common/rollout/workloads/statefulset_controller.go:51]: I reproduce this issue.I think this a test-side issue, in integration-test we generate a fake appRollout that not exist in k8s . so these fields are empty. You needn't modify it , i will fix it in my pr.

**Review by wangyikewxgm — APPROVED** [review]
LGTM

**Pull request merged**